src/UserCreator.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+namespace Icepace;
+
+use PDO;
+
+class UserCreator
+{
+    const USERNAME_REQUIRED = 'Please enter your username';
+    const USERNAME_TAKEN = 'Someone already has this username! Please enter a unique one';
+    const PASSWORD_REQUIRED = 'Please enter your password';
+    const PASSWORD_INVALID = 'Your password is shorter than 8 characters';
+    const BIO_REQUIRED = 'Please enter your bio';
+    const BIO_INVALID = 'Your bio is longer than 2000 characters';
+
+    private static function sanitiseUsername(string $username): string
+    {
+        return trim(filter_var($username, FILTER_SANITIZE_STRING));
+    }
+
+    private static function sanitiseBio(string $bio): string
+    {
+        return filter_var($bio, FILTER_SANITIZE_STRING);
+    }
+
+    private static function validateUsername(string $username): ?string
+    {
+        if ($username === '') {
+            return UserCreator::USERNAME_REQUIRED;
+        }
+        return null;
+    }
+
+    private static function validatePassword(string $password): ?string
+    {
+        if ($password === '') {
+            return UserCreator::PASSWORD_REQUIRED;
+        } elseif (strlen($password) < 8) {
+            return UserCreator::PASSWORD_INVALID;
+        }
+        return null;
+    }
+
+    private static function validateBio(string $bio): ?string
+    {
+        if ($bio === '') {
+            return UserCreator::BIO_REQUIRED;
+        } elseif (strlen($bio) > 2000) {
+            return UserCreator::BIO_INVALID;
+        }
+        return null;
+    }
+
```

Comment:
in order to be consistent with the rest of the file, it would be better in these functions to access the constants from `self::` rather than from `UserCreator::`

Reply:
fixed.